core/src/test/java/org/moditect/test/AddModuleInfoTest.java

```diff
@@ -117,25 +179,11 @@ private void prepareTestJar() throws Exception {
 
         Path exampleJar = GENERATED_TEST_RESOURCES.resolve( "example.jar" );
 
-
-//        Map<String, String> env = new HashMap<>();
-//        env.put( "create", "true" );
-//
-//        URI uri = URI.create( "jar:" + exampleJar.toUri().toString() );
-//
-//           try (FileSystem zipfs = FileSystems.newFileSystem(uri, env)) {
-//               Files.write( zipfs.getPath( "com/example/HelloWorld.class" ), ByteStreams.toByteArray( classFile.get().openInputStream()  ) );
-//            }
-//           catch(IOException e) {
-//                throw new RuntimeException( "Couldn't add module-info.class to JAR", e );
-//            }
-
-
-
-
-
         Manifest manifest = new Manifest();
         manifest.getMainAttributes().put( Attributes.Name.MANIFEST_VERSION, "1.0" );
+        if ( mrJar ) {
+            manifest.getMainAttributes().put( Attributes.Name.MULTI_RELEASE, "true" );
```

Comment:
Should we add this attribute to amended JARs (if not present yet) also in the actual Maven goal, if `jvmVersion` is given?

Reply:
Not sure what you mean. Could you please be more specific?

Reply:
Note that we can't always add this attribute because of https://bugs.openjdk.java.net/browse/JDK-8193802

Reply:
What I meant is, is this attribute needed for a JAR to be properly recognized as MR JAR? Or will the JVM also load class files from _META-INF/versions/..._ if that attribute is not present?

About the referenced JDK bug, I'm not quite clear what is triggering it?

Reply:
Ok, so the _add-module-info_ goal should set the attribute then if the `jvmVersion` is given and the patched JAR hasn't been an MR JAR before.

Reply:
Ok I'll add that in a second.